images/fcos-base-image/root/usr/share/bastion_services/registry/gen_registry_file.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+
+PORT="$1"
+registry_password_file="/opt/registry-${PORT}/auth/htpasswd"
+registry_certs_cert="/opt/registry-${PORT}/certs/domain.crt"
+registry_certs_key="/opt/registry-${PORT}/certs/domain.key"
+reg_cert_file="/usr/share/bastion_services/registry/domain.crt"
+reg_key_file="/usr/share/bastion_services/registry/domain.key"
+
+## GENERATE REGISTRY CONFIGY.YAML FILE
+cat > "/opt/registry-${PORT}/config.yaml" << EOF
+version: 0.1
+log:
+  fields:
+    service: registry
+http:
+  addr: :${PORT}
+  headers:
+    X-Content-Type-Options: [nosniff]
+  tls:
+    certificate: /certs/domain.crt
+    key: /certs/domain.key
+storage:
+  delete:
+    enabled: true
+  cache:
+    blobdescriptor: inmemory
+  filesystem:
+    rootdirectory: /var/lib/registry
+auth:
+  htpasswd:
+    realm: Registry Realm
+    path: /auth/htpasswd
+health:
+  storagedriver:
+    enabled: true
+    interval: 10s
+    threshold: 3
+EOF
+
+## REGISTRY PASSWORD
+echo 'dummy:$2y$05$7RzcQXdIGYLc6hI0luBGVuQGt2CB1WjFaX48JVgKBAt.BzSMZ/Q6G' > "${registry_password_file}"
```

Comment:
Can we consider this similarly to reg_cert_file? Assume you will have an htpasswd file in `${htpasswd_file}` (for example /opt/registry-common/htpasswd), you can copy that to `${registry_password_file}`.

The files in /opt/registry-common/ will be provided externally and the unit will be able to start and do the initial configuration.


Also add


```
ConditionPathExists=/opt/registry-common
```
in the `[unit]` section of the unit file: https://github.com/openshift-qe/baremetal-qe-infra/pull/13/files#diff-0471a597017ffe2f8d3aa15a1c6507a2e8b83dbdb703b2a959d1570bebee5ac2R5